primaryHandler.go

```diff
@@ -259,48 +259,58 @@ type handleWebhooksIn struct {
 
 type handleWebhooksOut struct {
 	fx.Out
+	AddWebhookHandler     http.Handler `name:"add_webhook_handler"`
+	GetAllWebhooksHandler http.Handler `name:"get_all_webhooks_handler"`
 }
 
-func handleWebhooks(in handleWebhooksIn) (out handleWebhooksOut, err error) {
-	//
-	// Webhooks (if not configured, handlers are not set up)
-	//
-	if in.V.IsSet(webhookConfigKey) {
-		webhookConfig := in.WebhookConfigKey
+type handleWebhookRoutesIn struct {
+	fx.In
+	Logger                log.Logger
+	APIRouter             *mux.Router `name:"api_router"`
+	Authenticate          *alice.Chain
+	AddWebhookHandler     http.Handler `name:"add_webhook_handler"`
+	GetAllWebhooksHandler http.Handler `name:"get_all_webhooks_handler"`
+}
 
-		webhookConfig.Logger = in.Logger
-		webhookConfig.MetricsProvider = in.MetricsRegistry
+func handleWebhookRoutes(in handleWebhookRoutesIn) {
+	in.APIRouter.Handle("/hook", in.Authenticate.Then(in.AddWebhookHandler)).Methods(http.MethodPost)
```

Comment:
We should check that `in.AddWebhookHandler != nil && in.GetAllWebhooksHandler != nil`.  If either are nil, we can return without setting any of this up.